### PR TITLE
`FidoEnumConverter`: fall back to deserializing by name even if `EnumMemberAttribute` is set

### PR DESCRIPTION
**Pull request by Tyrrrz**
Currently, `FidoEnumConverter` works in the following way:

1. If an enum value is annotated by `EnumMemberAttribute`, then the provided custom name is used both for serialization and deserialization.
2. If the attribute is absent, then the default name (i.e. field name) is used.

This PR adds another fallback layer:

3. If an enum value is annotated by `EnumMemberAttribute`, but the default name is provided (not custom) then the deserialization would still succeed. Serialization will work the same regardless.

In other words, it would correctly resolve `PublicKeyCredentialType.PublicKey` from both `public-key`, as well as `PublicKey`.

This makes it a bit easier for applications to use Fido2 types directly in their payloads without building their own converters, and without relying on the intricacies of Fido2 naming conventions.

### CODE REVIEW

**Comment by iamcarbon** [comment]
I had previously removed the relaxed parsing behaviors from this type, and believe they should remain strict. As a security critical library, we should continue to prioritize conformance with the specification 
over convenience.
